source/patterns/@aws-solutions-constructs/aws-s3-sns/test/test.s3-sns.test.ts

```diff
@@ -0,0 +1,235 @@
+
+test('All get methods return non-null objects', () => {
+  const stack = new Stack();
+  const app = new S3ToSns(stack, 'test-s3-sns', {
+
+  });
+  expect(app.snsTopic !== null);
+  expect(app.s3Bucket !== null);
+  expect(app.s3LoggingBucket !== null);
+  expect(app.encryptionKey !== null);
+  expect(app.s3BucketInterface !== null);
+});
+
+test('deployment works with existing bucket', () => {
+  const stack = new Stack();
+  const [ existingBucketObj ] = defaults.buildS3Bucket(stack, {});
+  new S3ToSns(stack, 'test-s3-sns', {
+    existingBucketObj
+  });
+
+  expect(stack).toHaveResource("Custom::S3BucketNotifications", {
+    NotificationConfiguration: {
+      TopicConfigurations: [
+        {
+          Events: ['s3:ObjectCreated:*'],
+          TopicArn: {
+            Ref: "tests3snsSnsTopicF02F6BD0"
+          }
+        }
+      ]
+    }
+  });
+});
+
+test('deployment works with existing topic and key', () => {
+  const stack = new Stack();
+  const cmk = defaults.buildEncryptionKey(stack);
+  const [ existingTopicObj ] = defaults.buildTopic(stack, {
+    encryptionKey: cmk
+  });
+  new S3ToSns(stack, 'test-s3-sns', {
+    existingTopicObj,
+    existingTopicEncryptionKey: cmk
+  });
+
+  expect(stack).toHaveResource("Custom::S3BucketNotifications", {
+    NotificationConfiguration: {
+      TopicConfigurations: [
+        {
+          Events: ['s3:ObjectCreated:*'],
+          TopicArn: {
+            Ref: "SnsTopic2C1570A4"
+          }
+        }
+      ]
+    }
+  });
+
+  expect(stack).toHaveResource("AWS::SNS::Topic", {
+    KmsMasterKeyId: {
+      "Fn::GetAtt": [
+        "EncryptionKey1B843E66",
+        "Arn"
+      ]
+    }
+  });
+});
+
+test('deployment works with specific evnet types and filters', () => {
+  const stack = new Stack();
+  new S3ToSns(stack, 'test-s3-sns', {
+    s3EventTypes: [ s3.EventType.OBJECT_REMOVED ],
+    s3EventFilters: [
+      {
+        prefix: 'the/place',
+        suffix: '*.mp3'
+      }
+    ]
+  });
+
+  expect(stack).toHaveResource("Custom::S3BucketNotifications", {
+    NotificationConfiguration: {
+      TopicConfigurations: [
+        {
+          Events: [
+            's3:ObjectRemoved:*'
+          ],
+          Filter: {
+            Key: {
+              FilterRules: [
+                {
+                  Name: 'suffix',
+                  Value: '*.mp3'
+                },
+                {
+                  Name: 'prefix',
+                  Value: 'the/place'
+                }
+              ]
+            }
+          },
+          TopicArn: {
+            Ref: "tests3snsSnsTopicF02F6BD0"
+          }
+        }
+      ]
+    }
+  });
+});
+
+test('Topic is encrypted with imported CMK when set on encryptionKey prop', () => {
+  const stack = new Stack();
+  new S3ToSns(stack, 'test-s3-sns', {
+    encryptionKey: new kms.Key(stack, 'cmk')
+  });
+
+  expect(stack).toHaveResource("AWS::SNS::Topic", {
+    KmsMasterKeyId: {
+      "Fn::GetAtt": [
```

Comment:
same comment - check not just that there's a key with an autogenerated name, use a unique attrib and look for that with hasResourceLike